### PR TITLE
Texturedata to data components

### PR DESCRIPTION
**Pull request by Nightenom**
Tried to make it backward compatible as much as possible - however MaterialTextureData is now trully immutable record
Is needed for loottables #243

### CODE REVIEW

File: src/api/java/com/ldtteam/domumornamentum/block/IMateriallyTexturedBlock.java [hunk 1]
```diff
@@ -69,7 +69,7 @@ default boolean isCorrectToolForDrops(BlockState state, final ItemStack stack, B
             {
                 return stack.isCorrectToolForDrops(state);
             }
-            Block block = mtbe.getTextureData().getTexturedComponents().get(getMainComponent().getId());
+            Block block = mtbe.getTextureData().texturedComponents().get(getMainComponent().getId());
```
Comment (Raycoms) [inline src/api/java/com/ldtteam/domumornamentum/block/IMateriallyTexturedBlock.java:72]: let's not rename this getter. This just creates a huge dif

Comment (Nightenom) [reply src/api/java/com/ldtteam/domumornamentum/block/IMateriallyTexturedBlock.java:72]: Agree, but Record issue, can name the record var as getter

Comment (someaddons) [reply src/api/java/com/ldtteam/domumornamentum/block/IMateriallyTexturedBlock.java:72]: just convert it to a class then

Comment (Nightenom) [reply src/api/java/com/ldtteam/domumornamentum/block/IMateriallyTexturedBlock.java:72]: DataComps should be strictly immutable class = record, or act as such due to vanilla, it was class before

Comment (Raycoms) [reply src/api/java/com/ldtteam/domumornamentum/block/IMateriallyTexturedBlock.java:72]: It's fine to be a class, just make stuff final. Or do they enforce record?

Comment (Nightenom) [reply src/api/java/com/ldtteam/domumornamentum/block/IMateriallyTexturedBlock.java:72]: As class it must have full data equals() so just like I said "to act as record" but can be mutable

Comment (Raycoms) [reply src/api/java/com/ldtteam/domumornamentum/block/IMateriallyTexturedBlock.java:72]: Probably easier to leave it a class then with final and equal method. records are always a pain fi we need sth else later too.

Comment (someaddons) [reply src/api/java/com/ldtteam/domumornamentum/block/IMateriallyTexturedBlock.java:72]: ye records are just syntax sugar, we can make an equivalent class with the same requirements, most IDE's even do the auto-conversion for you

Comment (Raycoms) [reply src/api/java/com/ldtteam/domumornamentum/block/IMateriallyTexturedBlock.java:72]: I'll merge this and do a followup PR with the change, as that's faster

Comment (Nightenom) [reply src/api/java/com/ldtteam/domumornamentum/block/IMateriallyTexturedBlock.java:72]: > ye records are just syntax sugar, we can make an equivalent class with the same requirements, most IDE's even do the auto-conversion for you

That's imho void argument, literally the point of syntax sugar is to make code more understandable

Comment (someaddons) [reply src/api/java/com/ldtteam/domumornamentum/block/IMateriallyTexturedBlock.java:72]: how is that a void argument when the point here is to keep the method name by using a record equivalent class?

Comment (Raycoms) [reply src/api/java/com/ldtteam/domumornamentum/block/IMateriallyTexturedBlock.java:72]: I also don't think records make code more understandable. They make it more compact.

**Pull request merged**